aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -15,220 +15,504 @@ uid: fundamentals/servers/kestrel
+* Maximum client connections
+* Maximum request body size
+* Minimum request body data rate
+
+Set these and other constraints on the `Limits` property of the [KestrelServerOptions](/dotnet/api/microsoft.aspnetcore.server.kestrel.core.kestrelserveroptions) class. The `Limits` property holds an instance of the [KestrelServerLimits](/dotnet/api/microsoft.aspnetcore.server.kestrel.core.kestrelserverlimits) class.
 
 **Maximum client connections**
 
-The maximum number of concurrent open TCP connections can be set for the entire application with the following code:
+[MaxConcurrentConnections](/dotnet/api/microsoft.aspnetcore.server.kestrel.core.kestrelserverlimits.maxconcurrentconnections)  
+[MaxConcurrentUpgradedConnections](/dotnet/api/microsoft.aspnetcore.server.kestrel.core.kestrelserverlimits.maxconcurrentupgradedconnections)
+
+The maximum number of concurrent open TCP connections can be set for the entire app with the following code:
 
-[!code-csharp[](kestrel/sample2/Program.cs?name=snippet_Limits&highlight=3-4)]
+[!code-csharp[](kestrel/samples/2.x/Program.cs?name=snippet_Limits&highlight=3-4)]
 
-There's a separate limit for connections that have been upgraded from HTTP or HTTPS to another protocol (for example, on a WebSockets request). After a connection is upgraded, it isn't counted against the `MaxConcurrentConnections` limit. 
+There's a separate limit for connections that have been upgraded from HTTP or HTTPS to another protocol (for example, on a WebSockets request). After a connection is upgraded, it isn't counted against the `MaxConcurrentConnections` limit.
 
 The maximum number of connections is unlimited (null) by default.
 
 **Maximum request body size**
 
-The default maximum request body size is 30,000,000 bytes, which is approximately 28.6MB. 
+[MaxRequestBodySize](/dotnet/api/microsoft.aspnetcore.server.kestrel.core.kestrelserverlimits.maxrequestbodysize)
+
+The default maximum request body size is 30,000,000 bytes, which is approximately 28.6 MB.
 
-The recommended way to override the limit in an ASP.NET Core MVC app is to use the [RequestSizeLimit](https://github.com/aspnet/Mvc/blob/rel/2.0.0/src/Microsoft.AspNetCore.Mvc.Core/RequestSizeLimitAttribute.cs) attribute on an action method:
+The recommended approach to override the limit in an ASP.NET Core MVC app is to use the [RequestSizeLimit](/dotnet/api/microsoft.aspnetcore.mvc.requestsizelimitattribute) attribute on an action method:
 
 ```csharp
 [RequestSizeLimit(100000000)]
 public IActionResult MyActionMethod()
 ```
 
-Here's an example that shows how to configure the constraint for the entire application, every request:
+Here's an example that shows how to configure the constraint for the entire app, every request:
```

Comment:
Should the ", every request" portion of this sentence be removed? Or maybe that should say ", on every request"?

Reply:
Let's drop "entire," too.

> Here's an example that shows how to configure the constraint for the app on every request: